splunk-otel-android/src/main/java/com/splunk/rum/CurrentNetwork.java

```diff
@@ -16,15 +16,30 @@
 
 package com.splunk.rum;
 
+import android.os.Build;
 import androidx.annotation.Nullable;
 import java.util.Objects;
 import java.util.Optional;
 
 final class CurrentNetwork {
+    @Nullable private final Carrier carrier;
     private final NetworkState state;
     @Nullable private final String subType;
 
+    CurrentNetwork(NetworkState state) {
+        this(state, null);
+    }
+
+    CurrentNetwork(Carrier carrier, NetworkState state) {
+        this(carrier, state, null);
+    }
+
     CurrentNetwork(NetworkState state, @Nullable String subType) {
+        this(null, state, subType);
+    }
+
+    CurrentNetwork(@Nullable Carrier carrier, NetworkState state, @Nullable String subType) {
```

Comment:
Nit: do we need a builder for this class?

Reply:
Yeah, might help actually with all the constructors. Lemme try it.